pkg/executor/wasm/handler.go

```diff
@@ -77,7 +82,37 @@ func (h *executionHandler) run(ctx context.Context) {
 		h.cancel()
 	}()
 
-	tracingEngine := tracedRuntime{h.runtime}
+	configuredProtocol := strings.ToLower(os.Getenv("OTEL_EXPORTER_OTLP_PROTOCOL"))
+	endpoint := os.Getenv("OTEL_EXPORTER_OTLP_ENDPOINT")
```

Comment:
I would much prefer we do this by re-using the config that Bacalhau is already providing rather than doing config parsing inline in the executor here.

Ideally you could just consume the [`telemetry.getTraceClient`](https://github.com/dylibso/bacalhau/blob/c7aa118c45d685bb7583330b8ba3b51bf12c483f/pkg/telemetry/tracing.go#L58-L75) and pass it to [`OTelAdapter.UseCustomClient`](https://github.com/dylibso/observe-sdk/blob/b4a765d1f1479c7a8b54d0939f571d28e86ae93e/go/adapter/opentelemetry/adapter.go#L41-L45). But @wikiwong prefers to not do this because it will tie your adapter to the same version of the OTel library that Bacalhau uses. 

For my understanding, isn't that a problem anyway? Can two versions of the same go lib sit side-by-side in the same binary? Won't we always need a match with the version of OTel used by dylibso?

So if we can't use the client directly, can we reuse more of the config options directly from our telemetry module? E.g. the [service  name](https://github.com/dylibso/bacalhau/blob/c7aa118c45d685bb7583330b8ba3b51bf12c483f/pkg/telemetry/common.go#L45), the [`OTEL_EXPORTER_OTLP_*`](https://github.com/dylibso/bacalhau/blob/c7aa118c45d685bb7583330b8ba3b51bf12c483f/pkg/telemetry/constants.go#L10) constants etc...?

Reply:
I don't think it would be a big issue to re-use the existing client, the function is already there so it seems worth testing this approach.